src/components/10-atoms/button/index.js

```diff
@@ -64,90 +56,25 @@ class AXAButton extends InlineStyles {
     return this.icon === ARROW_RIGHT;
   }
 
-  watch(mode) {
-    switch (mode) {
-      case 'stop':
-        if (this._observer) {
-          this._observer.disconnect();
-        }
-        break;
-      case 'start':
-        if (!this._observer) {
-          this._observer = new MutationObserver(() => this.attachFakeButton());
-        }
-        this._observer.observe(this, {
-          childList: true,
-        });
-        break;
-      default:
-        break;
+  handleClick(ev) {
+    if (this.type !== 'button' && hasShadowDom(this)) {
+      // this button wants to specifically submit a form
+      // climb up the dom to see if we're in a <form>
+      // and if so, then use JS to submit it
+      const form = this.closest('form');
+      if (form) {
+        ev.preventDefault();
+
+        const fakeButton = document.createElement('button');
```

Comment:
What about this alternative to lines  68-73: `form[this.type]();` ?!

Reply:
In whatever final solution you must link to https://www.tjvantoll.com/2013/01/01/enter-should-submit-forms-stop-messing-with-that/ in a comment.

The implicit-form-submission-via-enter algorithm causes internal synthetic click on a submit button if found. Said button cannot be discovered when hidden under shadowDOM. Therefore, your current solution is incorrect insofar as the fakeButton should _not_ be removed and that the fakeButton should be constructed in firstUpdated() since the algorithm can strike at any time, unconnected to a local click on this instance.

Reply:
An alternative is to _also_ modify axa-input-text: detect Enter keypress event, find closest form, submit  form (maybe with added conditions from the linked article).

Reply:
Another alternative is to rewrite button(-link) to use NoShadowDOM, and drop all complications like here. Isn't this the simplest,most HTML-conformant approach?